client/src/partials/purchases/create/create.html

```diff
@@ -0,0 +1,207 @@
+                  <textarea
+                    ng-model="PurchaseCtrl.order.details.note"
+                    name="notes"
+                    class="form-control"
+                    rows="4"
+                    ng-maxlength="PurchaseCtrl.maxLength"
+                    required>
+                  </textarea>
+                  <div class="help-block" ng-messages="PurchaseOrderForm.notes.$error" ng-show="PurchaseOrderForm.$submitted">
+                    <div ng-messages-include="partials/templates/messages.tmpl.html"></div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </form>
+        </div>
+      </div>
+    </div>
+
+
+    <!-- TODO Move padding to generic CSS class -->
+    <div class="row" style="padding-bottom : 5px;">
+      <div class="col-xs-12">
+
+        <!-- "Recover Items" button -->
+        <div class="btn-group pull-right" role="group">
+          <button
+            class="btn btn-default"
+            ng-class="{'btn-primary' : PurchaseCtrl.order.hasCacheAvailable() && PurchaseCtrl.order.details.supplier_uuid }"
+            ng-click="PurchaseCtrl.order.readCache()"
+            ng-disabled="!PurchaseCtrl.order.hasCacheAvailable() || !PurchaseCtrl.supplier">
+            <span class="fa fa-recycle"></span> {{ "FORM.BUTTONS.RECOVER_ITEMS" | translate }}
+          </button>
+        </div>
+
+        <!-- "Add number of grid rows" input-group -->
+        <div class="input-group">
+          <span class="input-group-btn">
+            <button
+              id="btn-add-rows"
+              class="btn btn-default btn-sm"
+              ng-disabled="!PurchaseCtrl.supplier"
+              ng-click="PurchaseCtrl.addItems(PurchaseCtrl.itemIncrement)">
+              <span class="fa fa-plus-circle"></span> {{ "FORM.BUTTONS.ADD" | translate }}
+            </button>
+          </span>
+          <input
+            type="number"
+            class="form-control input-sm"
+            ng-model="PurchaseCtrl.itemIncrement"
+            style="max-width : 40px;"
+            ng-disabled="!PurchaseCtrl.supplier">
+          </div>
+
+        <!-- @todo - hacky check for inventory to prevent local variables -->
+        <p
+          style="padding-top : 10px;"
+          class="text-warning"
+          ng-show="PurchaseCtrl.order.inventory.size() > 0 && PurchaseCtrl.order.inventory.available.data.length === 0">
+          <span class="fa fa-info-circle"></span>
+          {{ "FORM.INFOS.ITEMS_FULL" | translate }}
+        </p>
+
+        <p
+          style="padding-top : 10px;"
+          class="text-info"
+          ng-show="!PurchaseCtrl.supplier">
+          <span class="fa fa-info-circle"></span>
+          {{ "FORM.INFOS.NO_SUPPLIER" | translate }}
+        </p>
+      </div>
+    </div>
+
+    <div id="purchase-order-grid" ui-grid="PurchaseCtrl.gridOptions" style="height : 230px; width : 100%;" ui-grid-auto-resize ui-grid-cellNav>
```

Comment:
This would be a good candidate for `bhConstant` which landed in https://github.com/IMA-WorldHealth/bhima-2.X/pull/538 (After this pull request was submitted so entirely optional!) 

Something like: 
```html
<div style="height : {{::PurchaseCtrl.Constants.INVOICE.GRID_HEIGHT}} />
```